.github/workflows/pdm-lock-automation.yaml

```diff
@@ -75,17 +91,39 @@ jobs:
           else
             echo "Virtual environment already exists."
           fi
+
           source .venv/bin/activate
+
+          echo "Stashing changes if there are any..."
+          git stash || echo "No stash entries found, continuing..."
```

Comment:
@kirtimanmishrazipstack git stash should be done after pdm lock run anf before git pull.

Reply:
@ritwik-g done